### PR TITLE
WIP proposed template management of CMakeSettings.json

### PR DESCRIPTION
**Pull request by bencoman**
It will be good to provide source control ``CMakeSettings.json`` to provide standard configurations ,
but Visual Studio considers treats this as a user file so check-ins will have incidental noise. 
So instead use ``CMakeSettings.json.template`` to provide standard configurations
when this changes, CMake very early on copies the template to user ``CMakeSettings.json`` .
Discussion at #138.

### CODE REVIEW

**Comment by bencoman** [comment]
To facilitate operational testing with affecting existing ``CMakeSettings.json``, the code currently creates  ``CMakeSettings.jsonx``.

**Comment by bencoman** [comment]
HOLD review.  My further testing here has Cmake behaving annoying strangely.

**Comment by bencoman** [comment]
@guillep Before I put more effort into troubleshooting, could you skim the code and let me know your general thoughts on my concept - to have CMake automatically copy the ``CmakeSettings.json`` from a template, and also track when then template is updated.

**Comment by guillep** [comment]
> Before I put more effort into troubleshooting, could you skim the code and let me know your general thoughts on my concept - to have CMake automatically copy the `CmakeSettings.json` from a template, and also track when then template is updated.

I've taken a look, and I think we can try it. It will not break anything, and using it we will get some feedback. What do you think?

File: CMakeLists.txt [hunk 1]
```diff
@@ -15,6 +15,44 @@ set(VERSION_MAJOR 9)
...
+            message("NOTICE!! Your 'CMakeSettings.json' file has been replaced by updated template.") 
+            message("NOTICE!! It is recommended you base your configuration settings off this update.") 
+            message("NOTICE!! Your old settings saved to ${user_bak_file}")
+            message("NOTICE!! To force using your old settings, copy 'template-hash' from new 'CMakeSettings.json' into your restored file.") 
+            message(FATAL_ERROR "Not really an error, just an important NOTICE!! to read in output.") 
+        endif()
+    endif()
+endif()
+#help code review focus on this part by exiting quickly...
+message(FATAL_ERROR "EVERYTHING OKAY, JUST HELP QUICK CODE REVIEW")
```
Comment (bencoman) [inline CMakeLists.txt:52]: @guillep This line was left in while WIP.  I'll clean it up now.

P.S. Sorry for being slow to respond.  Work smashed me leading up to xmas. 
 I have a couple of weeks free now to look further into VM on FreeRTOS.

**Pull request merged**